packages/components/MIGRATING.md

```diff
@@ -66,6 +69,105 @@ Changes:
 </Label>
 ```
 
+### Dialog
+
+Changes:
+
+-   The `Dialog` component now controls its open state internally by default.
+
+    -   You can pass in a trigger component as a child of `Dialog.Trigger` to control the open state.
+
+    -   The `open` and `onOpenChange` props can still be used to control the open state externally if needed.
+
+-   The `Dialog` now provides multiple subcomponents.
+
+    -   _required_ - The `Dialog.Root` wraps all other subcomponents and handles the dialog state and modality.
+
+    -   _required_ - The `Dialog.Trigger` is used to pass in a component to trigger the dialog.
+
+    -   _required_ - The `Dialog.Content` is the container appearing when the dialog is Visible
+
+        -   _optional_ - The `Dialog.Header` can be passed in as a child of `Dialog.Content` to add a styled header to the dialog.
+
+        -   _optional_ - The `Dialog.Body` can be passed in as a child of `Dialog.Content` to add a styled body to the dialog.
+
+        -   _optional_ - The `Dialog.Footer` can be passed in as a child of `Dialog.Content` to add a styled footer to the dialog.
+
+        -   _optional_ - The `Dialog.SideContent` can be passed in as a child of `Dialog.Content` to add a space on the left side of the dialog. Custom components can then be passed in as as children of `Dialog.SideContent`
+
+-   The styling / layout is now controlled on the subcomponent `Dialog.Content`
+
+-   The subcomponent `Dialog.Content` is used to display the dialog container.
+    It provides no styling by default and can be used to wrap a custom content.
+
+    -   `roundedCorners` is now called `rounded` and is a boolean.
+
+    -   The `width` and `maxHeight` props were removed. The Dialog container adjusts to the content inside. You can use a custom component inside if neccessary. you can use `minWidth`, `maxWidth` and `minHeight` to override the default values.
+
+    -   The `strategy` and `role` props were removed to simplify the API.
+
+    -   The `enablePortal` and `verticalAlignment` props were removed to standardize the dialog.
+
+        The dialog now uses a portal by default and is placed centered
+
+-   The subcomponents `Dialog.Header`, `Dialog.Body`, `Dialog.Footer` and `Dialog.SideContent` can be used to add defaut styling to the content inside of `Dialog.Content`.
+
```

Comment:
Please talk about removal of `darkUnderlay`, `autoHeight` and `anchor` and their replacement if any

Reply:
What anchor prop do you mean?

The rest is now added

Reply:
<img width="511" alt="Screenshot 2024-05-27 at 17 16 17" src="https://github.com/Frontify/fondue/assets/6846950/9dad343e-81c7-48d2-a249-ec7493258d61">

Reply:
Not so sure about that, `anchor` is a required prop

Reply:
added a comment, even though it wasn't documented before.